### PR TITLE
Fix isEmoji Function

### LINKED ISSUE #15: Smile classifies an asterisk as an emoji
When using smile to detect emoji for "*" it returns that there is an emoji present.

### LINKED ISSUE #22: isEmoji function is returning false for alternate style emojis
I have tried several alternate style emojis and they are all returning false in isEmoji. I tried several and any alternate styles I tried would return false. A couple I know I tried are 🤴🏻, 🦹🏻‍♀️

### PR DESCRIPTION
**Pull request by justinkumpe**
This PR updates the isEmoji function to check containsEmoji, containsOnlyEmoji, and isSingleEmoji in one function. This also resolves #22 where isEmoji returned false on several alternate emojis and resolves #15 where isEmoji would return true for an asterisk.

Also updated readme file to account for these changes.

Updated podsec file to version 2.0.2. If this pull request is merged please also push new version to CocoaPods.

### CODE REVIEW

**Comment by onmyway133** [comment]
@justinkumpe Hi Justin, thanks for the PR 👍 What do you think of introducing some method on Smile, like Smile.isSingleEmoji, Smile.is2DigitsEmoji, Smile.isEmojiOnly, ... instead of extending String? And would be great if there are some tests to ensure the correctness

**Review by justinkumpe — COMMENTED** [review]
@onmyway133 Try this one. Introduced Smile.countEmoji, Smile.countNonEmoji, Smile.isSingleEmoji, and Smile.isEmojiOnly.

File: Sources/Smile.swift [hunk 1]
```diff
@@ -233,3 +295,31 @@ fileprivate extension String {
     return components(separatedBy: .decimalDigits).joined()
   }
 }
+
+// MARK: - emojiRules
+
+/// Run supplied emojiRule against string
+@available(iOS 10.2, *)
+public func checkEmojiRule(emojiRule: emojiRule, string: String) -> Any{
```
Comment (onmyway133) [inline Sources/Smile.swift:303]: What do you think of omitting this function? I guess users can use the methods isEmojiOnly, countNonEmoji directly? This method returns Any and introduces another enum `emojiRule

**Comment by onmyway133** [comment]
@justinkumpe looks a lot better now, 👍

File: Sources/Smile.swift (hunk 1 above)
Comment (justinkumpe) [reply Sources/Smile.swift]: Agreed. I was using this function to assist in testing the rules easier. Not sure why I decided to leave it in there.

**Comment by justinkumpe** [comment]
@onmyway133 Removed the checkEmojiRule function. Not sure exactly why I decided to leave it in there to begin with.

**Comment by onmyway133** [comment]
@justinkumpe This is awesome, thanks 👍 I guess I need to change the indentation from 2 to 4 spaces. I started working in places where 4 spaces is preferred and I get used to that now.

**Pull request merged**